cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -235,6 +235,21 @@ template <typename T = uint8_t>
 {
   // Transfer chunk data, coalescing adjacent chunks
   std::vector<std::future<size_t>> read_tasks;
+  size_t compressed_read_size   = 0;
+  size_t uncompressed_read_size = 0;
+
+  struct read_info {
+    size_t io_size;
+    size_t io_offset;
+    bool compressed;
+    bool device_read;
+    size_t begin_chunk;
+    size_t end_chunk;
+  };
```

Comment:
Wow, I thought that defining struct/class inside a function is not permitted in C++?

Reply:
It is permitted, but with some limitations (e.g. no static data members).